homeassistant/components/light/limitlessled.py

```diff
@@ -165,6 +169,22 @@ def __init__(self, group, config):
         self._brightness = None
         self.config = config
 
+    @asyncio.coroutine
+    def async_added_to_hass(self):
+        """Restore the state."""
+        old_state = yield from async_get_last_state(self.hass, self.entity_id)
+
+        if not old_state:
+            return
+
+        params = {key: old_state.attributes[key] for key in PROP_TO_ATTR
+                  if key in old_state.attributes}
+
+        # We need to call turn_on here because the component turns the devices
+        # off on init to achieve a known state.
+        if old_state.state == STATE_ON:
+            yield from self.async_turn_on(**params)
```

Comment:
You can't call a `action` on `async_added_to_hass`. You can do is like `self._state = STATE_ON`

Reply:
But wouldn't this only set the internal state of the component? The Problem is, that I actually have to emit the old state to the bulbs as the component changes their configuration in its init.

Reply:
In this case, that is the wrong way.

The restore function is only to set internal stuff and not to reset external things. I think we can close this now.

Maybe a snapshot component they can snapshot states and set it with a service will be the best way. So you can use this services inside automation/scripts to restore things to old state but that will be complex.